checker/tests/resourceleak/MustCallAliasNormalExit.java

```diff
@@ -0,0 +1,73 @@
+// Test case for https://github.com/typetools/checker-framework/issues/5597
+
+import java.io.*;
+import org.checkerframework.checker.calledmethods.qual.*;
+import org.checkerframework.checker.mustcall.qual.*;
+
+@InheritableMustCall("close")
+public class MustCallAliasNormalExit {
+
+  final @Owning InputStream is;
+
+  @MustCallAlias MustCallAliasNormalExit(@MustCallAlias InputStream p, boolean b) throws Exception {
+    if (b) {
+      throw new Exception("an exception!");
+    }
+    this.is = p;
+  }
+
+  @EnsuresCalledMethods(value = "this.is", methods = "close")
+  public void close() throws IOException {
+    this.is.close();
+  }
+
+  public static @MustCallAlias MustCallAliasNormalExit mcaneFactory(InputStream is)
+      throws Exception {
+    return new MustCallAliasNormalExit(is, false);
+  }
+
+  // :: error: required.method.not.called
+  public static void testUse1(@Owning InputStream inputStream) throws IOException {
+    MustCallAliasNormalExit mcane = null;
+    try {
+      mcane = new MustCallAliasNormalExit(inputStream, true); // at run time, this WILL throw
+    } catch (Exception e) {
+      // At run time would fail (NPE), but for illustrative purposes this is fine.
+      // This absolutely must not cause inputStream to be considered closed because of the
+      // MustCallAlias
+      // relationship.
+      mcane.close();
+    }
+  }
+
+  // :: error: required.method.not.called
+  public static void testUse2(@Owning InputStream inputStream) throws IOException {
+    MustCallAliasNormalExit mcane = null;
+    try {
+      mcane = mcaneFactory(inputStream);
+    } catch (Exception e) {
+      mcane.close();
+    }
+  }
+
+  // :: error: required.method.not.called
+  public static void testUse3(@Owning InputStream inputStream) throws Exception {
+    // if mcaneFactory throws, then inputStream goes out of scope w/o being closed
+    MustCallAliasNormalExit mcane = mcaneFactory(inputStream);
+    mcane.close();
+  }
+
+  // TODO: this appears to be a false positive, but the RLC doesn't handle it correctly because
```

Comment:
Should we open a separate issue for this one?  Looks weird.

Reply:
Up to you. I'm not actually that surprised that we issue a false positive on this code, because the chain of reasoning is a bit confusing:
* if `mcaneFactory` does throw an exception, then `close()` is directly called on `inputStream` and no must-call-alias relationship is established
* if `mcaneFactory` doesn't throw an exception, then `mcane` and `inputStream` are resource aliases, and their obligations are fulfilled via `mcane.close()` on line 71

I think the problem for the consistency checker here is that the alias sets are different on the different paths, but I'm not sure. So, maybe we should open an issue.

Reply:
Regardless, I'm going to merge this PR.

Reply:
> I think the problem for the consistency checker here is that the alias sets are different on the different paths, but I'm not sure. So, maybe we should open an issue.

As I understand it, the consistency checker does not do any kind of merge / least-upper bound computation at control-flow merges.  It should handle each of the paths separately.  So I am surprised by this behavior.  I can open an issue.